Framework/Algorithms/inc/MantidAlgorithms/Q1DWeighted.h

```diff
@@ -77,18 +77,35 @@ class MANTID_ALGORITHMS_DLL Q1DWeighted : public API::Algorithm {
   void calculate(const API::MatrixWorkspace_const_sptr &);
   void finalize(const API::MatrixWorkspace_const_sptr &);
 
+  void getTableShapes();
+  void getViewportParams(std::string &,
+                         std::map<std::string, std::vector<double>> &);
+  void getSectorParams(std::vector<std::string> &,
+                       std::map<std::string, std::vector<double>> &);
+  bool checkIfSymetricalWedge(double innerRadius, double outerRadius,
+                              double centerX, double centerY,
+                              double centerAngle, double angleRange);
   std::vector<std::vector<std::vector<double>>> m_intensities;
   std::vector<std::vector<std::vector<double>>> m_errors;
   std::vector<std::vector<std::vector<double>>> m_normalisation;
   std::vector<double> m_qBinEdges;
   size_t m_nQ;
   size_t m_nLambda;
   size_t m_nWedges;
+
+  std::vector<double> m_wedgesInnerRadius;
```

Comment:
This makes a lot of vectors. Perhaps it'd be better to define a struct for each sector that has all the properties, and have one vector of those structs.